### PR TITLE
BaSP-MR-59: Class add and edit form

### PR DESCRIPTION
**Pull request by EmaBaldor**
Added add and edit Classes form.
I worked in the Classes list and added new functions of edit and add new Class.

### CODE REVIEW

**Comment by EmaBaldor** [comment]
[[Class] Add/Edit Form](https://trello.com/c/ICY2bqGT/59-class-add-edit-form)

File: src/Components/Classes/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    const fetchTrainers = async () => {
+      try {
+        const response = await fetch(process.env.REACT_APP_API_URL + '/trainers');
+        const data = await response.json();
+        setTrainers(data.data);
+      } catch (error) {
+        console.error('Error fetching trainers:', error);
+      }
+    };
```
Comment (fernandofernandezuy) [inline src/Components/Classes/Form/index.jsx]: I think you could declare this variables outside the useEffect

File: src/Components/Classes/index.jsx [hunk 2]
```diff
@@ -1,55 +1,142 @@
...
+      <div className={styles.modalOverlay}>
+        <div className={styles.modalContent}>
+          <p>{modalMessage}</p>
+          <button className={styles.modalButton} onClick={closeModal}>
+            Close
+          </button>
+        </div>
+      </div>
+    );
+  };
```
Comment (fernandofernandezuy) [inline src/Components/Classes/index.jsx]: This modal shouldn't be declared here, you could declare outside this index and export it

File: src/Components/Classes/index.jsx [hunk 3]
```diff
@@ -1,55 +1,142 @@
...
-      />
-      <Table data={classes} deleteItem={deleteItem} setShowModal={setShowModal} />
+      {showForm ? (
+        <>
+          {editMode ? <h2>Edit Class</h2> : <h2>Add New Class</h2>}
+          <ClassesForm
+            onSubmit={handleSubmit}
+            onCancel={handleCancel}
+            editMode={editMode}
+            classe={selectedClasses}
```
[the remark this answers is not in the document]
Comment (AleMac17) [reply src/Components/Classes/index.jsx:127]: asdas

[the remark this answers is not in the document]
Comment (AleMac17) [reply src/Components/Classes/index.jsx:127]: 'class' is not allowed as a variable declaration name, that's why the 'classe' name.

[the remark this answers is not in the document]
Comment (EmaBaldor) [reply src/Components/Classes/index.jsx:127]: 'class' is not allowed as a variable declaration name, that's why the 'classe' name.

File: src/Components/Classes/Form/index.jsx (hunk 1 above)
Comment (EmaBaldor) [reply src/Components/Classes/Form/index.jsx:42]: Resolved.

File: src/Components/Classes/index.jsx (hunk 2 above)
Comment (EmaBaldor) [reply src/Components/Classes/index.jsx]: Resolved. modalClose component has been imported from the Modal folder.

File: src/Components/Classes/Form/index.jsx (hunk 1 above)
Comment (EmaBaldor) [reply src/Components/Classes/Form/index.jsx]: Resolved.

**Review by fernandofernandezuy — APPROVED** [review]
Looks good to me!

**Review by LucianoAlarcon — APPROVED** [review]

**Pull request merged**